src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,464 @@
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Program to collect error metrics on bases stratified in various ways.
+ *
+ * @author  Yossi Farjoun
+ */
+
+ @CommandLineProgramProperties(
+        summary = "Program to collect error metrics on bases stratified in various ways." +
+                "\n" +
+                "Sequencing errors come in different 'flavors', for example some occur during sequencing while " +
+                "others happen during 'library construction'. They can be correlated with various aspect of " +
+                "the sequencing experiment: position in the read, base context, length of insert and so on. " +
+                "" +
+                "This program enables the user to collect two different kinds of error metrics (one which attempts to " +
+                "distinguish between pre- and post- sequencer errors, and on which doesn't) and a collation of 'stratifiers' " +
+                "each of which assigns bases into various bins. To complicate things further, the stratifiers can be used together " +
+                "to generate a composite stratification. " +
+                "" +
+                "For example:\n" +
+                "The BASE_QUALITY stratifier will place bases in bins according to their declared base quality. " +
+                "The READ_ORDINALITY stratifier will place bases in one of two bins depending on whether their read " +
+                "is 'first' or 'second'. One could generate a composite stratifier BASE_QUALITY,READ_ORDINALITY which will " +
+                "do both as the same time. " +
+                "" +
+                "The resulting metric file will be named according to a provided prefix and a suffix which is generated " +
+                " automatically according to the error metric. " +
+                "To save on run-time, the tool collects many metrics in a single pass and there should be hardly any " +
+                "performance loss when collecting many metrics at the same time, which is why the default already includes a " +
+                "lengthy collection of metrics. " +
+                "" +
+                "The tool takes in a VCF for the sample " +
+                "to skip actually polymorphic bases. As a result it assumes that any disagreement from the reference is an error." +
+                "" +
+                "This assumption can, of-course be wrong, but the rates at which it is wrong is likely to be lower than" +
+                "the error rates that are exposed. " +
+                "" +
+                "The program also takes a region over which to run. This region should be used to define a 'confidence region' " +
+                "where lack of a call in the VCF translates to a high probability of a monomorphic site.",
```

Comment:
I don't see the connection to the VCF as described.  The VCF is just used to skip sites that are assumed to have non-reference alleles, while the regions over which to run are simply that.  The latter region _could_ be generated from the target-capture regions, or creating a set of regions with high coverage, or looking at the gVCF, or just the NIST high-confidence regions, for a start, so I think it is confusing to have the second sentence (I was confused).